client/ios-swift/app/CheckoutViewController.swift

```diff
@@ -0,0 +1,130 @@
+    }()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .white
+        let stackView = UIStackView(arrangedSubviews: [cardTextField, payButton])
+        stackView.axis = .vertical
+        stackView.spacing = 20
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(stackView)
+        NSLayoutConstraint.activate([
+            stackView.leftAnchor.constraint(equalToSystemSpacingAfter: view.leftAnchor, multiplier: 2),
+            view.rightAnchor.constraint(equalToSystemSpacingAfter: stackView.rightAnchor, multiplier: 2),
+            stackView.topAnchor.constraint(equalToSystemSpacingBelow: view.topAnchor, multiplier: 2),
+        ])
+        loadPage()
+    }
+
+    func displayAlert(title: String, message: String, restartDemo: Bool = false) {
+        DispatchQueue.main.async {
+            let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+            if restartDemo {
+                alert.addAction(UIAlertAction(title: "Restart demo", style: .cancel) { _ in
+                    self.cardTextField.clear()
+                    self.loadPage()
+                })
+            }
+            else {
+                alert.addAction(UIAlertAction(title: "OK", style: .cancel))
+            }
+            self.present(alert, animated: true, completion: nil)
+        }
+    }
+
+    func loadPage() {
+        // Create a SetupIntent by calling the sample server's /create-setup-intent endpoint.
+        let url = URL(string: BackendUrl + "create-setup-intent")!
+        var request = URLRequest(url: url)
+        request.httpMethod = "POST"
+        let task = URLSession.shared.dataTask(with: request, completionHandler: { [weak self] (data, response, error) in
+            guard let response = response as? HTTPURLResponse,
+                response.statusCode == 200,
+                let data = data,
+                let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String : Any],
+                let clientSecret = json["clientSecret"] as? String,
+                let stripePublishableKey = json["publishableKey"] as? String else {
+                    let message = error?.localizedDescription ?? "Failed to decode response from server."
+                    self?.displayAlert(title: "Error loading page", message: message)
+                    return
+            }
+            print("Created SetupIntent")
+            self?.setupIntentClientSecret = clientSecret
+            Stripe.setDefaultPublishableKey(stripePublishableKey)
+        })
+        task.resume()
+    }
+
+    @objc
+    func pay() {
+        guard let setupIntentClientSecret = setupIntentClientSecret else {
+            return;
+        }
+        // Collect card details
+        let cardParams = cardTextField.cardParams
+        let paymentMethodParams = STPPaymentMethodParams(card: cardParams, billingDetails: nil, metadata: nil)
```

Comment:
@adreyfus-stripe – curious how you feel about collecting email for the "saving cards" sample and passing to `billing_details`, like we do in the web sample [0]. Not sure what the benefit is to passing an email in the save card flow (I thought it had something to do with creating a Customer, but it doesn't look like that's happening in the web sample). 

[0] https://github.com/stripe-samples/saving-card-without-payment/blob/master/client/script.js

Reply:
We actually do use it to save the Customer! e.g. https://github.com/stripe-samples/saving-card-without-payment/blob/master/server/node/server.js#L87

I figured email would be one way developers could map Stripe Customers to their own internal representation of a Customer

Reply:
I'm happy to add a field for email and pass it.  

It seems like there are many ways to figure out what Customer the PM belongs to, and we don't have a particular recommendation.  It's required to attach the PaymentMethod to a Customer, but I think the sample also works if we leave the *how* out of scope, with a comment like `// Include e.g. the customer’s email, account id, Stripe Customer id, etc. so you know which customer the PaymentMethod belongs to.`

@adreyfus-stripe - also, I ran into some unexplained steps in the instructions here (can't find it in the latest web repo either):
* How do I get the `STRIPE_WEBHOOK_SECRET`?
* How do I point Stripe to my local `/webhook` endpoint?  

ended up using the Stripe CLI, which answered both questions.  Should we document that?